android/src/org/kiwix/kiwixmobile/LibraryFragment.java

```diff
@@ -103,6 +105,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         noNetworkConnection();
       }
 
+      this.getContext().registerReceiver(new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+
+          permissionButton.setVisibility(View.GONE);
+          permissionText.setVisibility(View.GONE);
+          progressBar.setVisibility(View.INVISIBLE);
+          progressBarLayout.setVisibility(View.GONE);
+          progressBarMessage.setVisibility(View.GONE);
+
+          NetworkInfo network = conMan.getActiveNetworkInfo();
+
+          if (network != null && network.isConnected()) {
+            if (books == null) {
+              if (isWiFi()) {
+                getLibraryData();
+              } else {
+                displayNetworkConfirmation();
+              }
+            }
+          } else {
+            books = null;
```

Comment:
When user looses internet and tries to download a book app crashes (#413).
But maybe it would be a better solution to show toast that there is no internet connection when user tries to download book without internet. What do you think?